### PR TITLE
Persist Yearly Reading Goals banner preferences

### LINKED ISSUE #8814: Don't show the Yearly Reading Goal banner every login
Every time I log in, I see the aforementioned banner. I think after using "x" to remove it once or twice, it shouldn't keep reappearing, yet it does.

### Describe the problem that you'd like solved
Limit the number of times the Yearly Reading Goal banner appears upon logging in.

### Proposal & Constraints
It doesn't have to be just once, but I don't think it should be every time until the end of February (or whatever time it's set to end).

### PR DESCRIPTION
**Pull request by QuantuM410**
Closes #8814 

### Technical

This PR addresses the issue where the Yearly Reading Goal banner reappears upon each login. The solution involves persisting the banner preference on the User, preventing the banner from displaying every time a user logs in.

### Testing

- Dismiss the "Your Reading Goal" banner 
- You can monitor browser storage for the respective cookie starting with "yrg" stored locally
- Remove the cookies or open another browser
- Log in and check the "yrg" cookies being stored as per the user preference leading to no display of the YRG banner

### Stakeholders

@jimchamp

### CODE REVIEW

File: openlibrary/plugins/upstream/account.py [hunk 1]
```diff
@@ -348,6 +348,15 @@ def POST(self):
...
                 if ol_account and ol_account.get_user().get_safe_mode() == 'yes':
                     web.setcookie('sfw', 'yes', expires=expires)
+                if (
+                    ol_account
+                    and 'yrg_banner_pref' in ol_account.get_user().preferences()
+                ):
+                    web.setcookie(
+                        ol_account.get_user().preferences()['yrg_banner_pref'],
+                        '1',
+                        expires=(3600 * 24 * 30),
```
Comment (jimchamp) [inline openlibrary/plugins/upstream/account.py:358]:
```suggestion
                        expires=(3600 * 24 * 365),
```
This shouldn't expire for one year.

File: openlibrary/plugins/upstream/account.py [hunk 2]
```diff
@@ -414,6 +423,12 @@ def POST(self):
         ol_account = OpenLibraryAccount.get(email=email)
         if ol_account and ol_account.get_user().get_safe_mode() == 'yes':
             web.setcookie('sfw', 'yes', expires=expires)
+        if ol_account and 'yrg_banner_pref' in ol_account.get_user().preferences():
+            web.setcookie(
+                ol_account.get_user().preferences()['yrg_banner_pref'],
+                '1',
+                expires=(3600 * 24 * 30),
```
Comment (jimchamp) [inline openlibrary/plugins/upstream/account.py:430]:
```suggestion
                expires=(3600 * 24 * 365),
```

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks, @QuantuM410!

Just a few small changes and this will be ready to merge.

**Comment by QuantuM410** [comment]
@jimchamp I have updated the code with the required changes, Thanks :)

**Review by jimchamp — APPROVED** [review]
Lgtm! Thanks, @QuantuM410

**Pull request merged**